### PR TITLE
Add og tags for better social media preview 

### PR DESCRIPTION
**Pull request by vipulgupta2048**
Solves #269 adds og tags to base HTML file 

Also, 
- Removed unicode literals from pelicanconf.py (Python 3 has them default)
- ~Need to test this since the `og:image` didn't work while testing with `og preview` extension.~ Works now.
- Created this banner to be `og:image` 

opengraph-default

@pythonindia/pycon-india-2020-infrastructure @pythonindia/pycon-india-content-team

### CODE REVIEW

**Comment by vipulgupta2048** [comment]
Reviews needed folks before merging.  
@sayanchowdhury @palnabarun @SaptakS

File: themes/inpycon2020/templates/base.html [hunk 1]
```diff
@@ -10,6 +10,19 @@
...
+    <meta property="og:url" content="{{ SITEURL }}/{{ article.url }}" />
+    <meta property="og:title" content="{{ article.title | replace("\"", "&quot;") }}" />
+    <meta property="og:description" content="{{ article.content | striptags | replace("\"", "&quot;") | truncate(124, False, '...') }}" />
+    <meta property="og:image" content="https://in.pycon.org/blog/theme/images/opengraph-default.png" />
+    {% else %}
+    <meta property="og:type" content="website" />
+    <meta property="og:url" content="{{ SITEURL }}/{{ output_file }}" />
+    <meta property="og:title" content="{{ SITENAME }}" />
+    <meta property="og:description" content="{{ DESCRIPTION }}" />
+    <meta property="og:image" content="https://in.pycon.org/blog/theme/images/opengraph-default.png" />
```
Comment (ananyo2012) [inline themes/inpycon2020/templates/base.html:24]: Does using SITEURL here cause any issue?

Comment (vipulgupta2048) [reply themes/inpycon2020/templates/base.html:24]: None that I have seen yet while testing.

Comment (ananyo2012) [reply themes/inpycon2020/templates/base.html:24]: Ah I meant why why is SITEURL not used in `og:image` ?

**Comment by palnabarun** [comment]
@vipulgupta2048 Please rebase against `master` and resolve the conflicts. :slightly_smiling_face:

File: themes/inpycon2020/templates/base.html [hunk 2]
```diff
@@ -10,6 +10,19 @@
     <link rel="shortcut icon" href="/blog/theme/images/favicon.png"/>
     <link rel="stylesheet" href="/blog/theme/css/bootstrap.min.css"/>
     <link rel="stylesheet" href="/blog/theme/css/styles.css"/>
+    {% if article is defined %}
+    <meta property="og:type" content="article" />
+    <meta property="og:url" content="{{ SITEURL }}/{{ article.url }}" />
+    <meta property="og:title" content="{{ article.title | replace("\"", "&quot;") }}" />
+    <meta property="og:description" content="{{ article.content | striptags | replace("\"", "&quot;") | truncate(124, False, '...') }}" />
+    <meta property="og:image" content="https://in.pycon.org/blog/theme/images/opengraph-default.png" />
```
Comment (palnabarun) [inline themes/inpycon2020/templates/base.html:18]: Use `{{ SITEURL }}` here plus the resource (theme/images/opengraph-default.png) doesn't exist as well as not added in this PR.

Am I missing something here? @vipulgupta2048

Comment (palnabarun) [inline themes/inpycon2020/templates/base.html:18]: Use `{{ SITEURL }}` here plus the resource (theme/images/opengraph-default.png) doesn't exist as well as not added in this PR.

Am I missing something here? @vipulgupta2048

Comment (vipulgupta2048) [reply themes/inpycon2020/templates/base.html:18]: No worries. Resolving!

File: themes/inpycon2020/templates/base.html (hunk 1 above)
Comment (vipulgupta2048) [reply themes/inpycon2020/templates/base.html:24]: Open Graph tags are not used by the blog itself, but by social media websites like FB or Twitter. Websites check the `og:image` tag to find the absolute path of the image.

With the use of `SITEURL` the path is `/blog/theme/images/opengraph-default.png` which is relative and breaks the image on the preview. Hope this helps in understanding. Resolving.

**Comment by vipulgupta2048** [comment]
Houston, we are ready for launch! :rocket: 
cc: @palnabarun

**Comment by ananyo2012** [comment]
Got it. Maybe we can update the same image as the website uses once the
website is updated.

Ship it !

**Comment by palnabarun** [comment]
> Houston, we are ready for launch! rocket

@vipulgupta2048 I would have preferred Sri Harikota, but fine. :man_shrugging:

**Comment by palnabarun** [comment]
@vipulgupta2048 Can you run `git rebase master` on this branch please? Merge commits look very ugly.

**Comment by vipulgupta2048** [comment]
Rebased!

**Comment by palnabarun** [comment]
@vipulgupta2048 Thanks! :+1:

**Pull request merged**